scanners/seo.py

```diff
@@ -106,26 +153,39 @@ def scan(domain: str, environment: dict, options: dict) -> dict:
             if description:
                 descriptions.append(description['content'])
             # and can we find dc:date?
-            dc_date = htmlsoup.select_one("meta[name='DC.Date']")
+            dc_date = htmlsoup.select_one("meta[name='article:published_time']")
+            if not dc_date:
+                dc_date = htmlsoup.select_one("meta[name='article:modified_time']")
+                if not dc_date:
+                    dc_date = htmlsoup.select_one("meta[name='DC.Date']")
             # if we found one, grab the content
             if dc_date:
                 dc_date = dc_date['content']
 
-            # Find the main tag (or alternate)
+            # Find the main tag (or alternate), if we haven't found one already.
             # Potential TO-DO: check that there is only one. Necessary? ¯\_(ツ)_/¯
-            maintag = True if htmlsoup.find('main') else False
-            # if we couldn't find `main` look for the corresponding role
-            if not maintag:
-                maintag = True if htmlsoup.select('[role=main]') else False
-
-            results[page] = {
-                'page': page,
-                'status': str(r.status_code),
-                'title': title,
-                'description': description,
-                'has main tag': maintag,
-                'DC Date': dc_date
-            }
+            if not results['Main tags found']:
+                maintag = True if htmlsoup.find('main') else False
```

Comment:
Style: I think you can simplify these to `bool(htmlsoup.find('main'))` etc..

Reply:
Simpler, yes, but at least in my opinion the way I have it is more explicit and clearer.